### ISSUE TITLE
How to handle optimizer state over multiple triggers?

### ISSUE THREAD

**Issue by MaxiBoether** [eth-easl/modyn#137]
It is not exactly clear right now how the optimizer state should be handled over multiple triggers. This might also be a benchmarking/evaluation question, but we should also mathematically think about this.

Imagine that we have a SGD optimizer with momentum, i.e., we reduce the learning rate with each `optimizer.step()`. If we forget about triggers, then we would continually reduce the learning rate. If we now introduce triggers and continue to finetune on the previous model, we would (with the vanilla approach), reset the optimizer because it is a new training job. That would mean that we suddenly change weights more radically again. 

On the one hand, this does not reflect the local one training setting. On the other hand, we might even want this, because the samples are newer and should change our model weights more.

**Comment by MaxiBoether** [comment]
Note that currently optimizer state is probably (without us remembering) stored in the checkpoint. We need to validate this and make this clean
